packages/framework/aqueduct/api-report/aqueduct.api.md

```diff
@@ -84,6 +85,9 @@ export interface ContainerRuntimeFactoryWithDefaultDataStoreProps {
     runtimeOptions?: IContainerRuntimeOptions;
 }
 
+// @internal
+export function createDataObjectKind<T extends new (...any: any[]) => DataObject>(factory: T): T & SharedObjectKind<InstanceType<T>>;
```

Comment:
This Unified the APi shape/types for data objects in the declarative API, avoiding adding any extra typing to support them.